Library/Homebrew/cmd/tap.rb

```diff
@@ -56,4 +62,8 @@ def tap
   def full_clone?
     ARGV.include?("--full") || ARGV.homebrew_developer?
   end
+
+  def force_auto_update?
+    true if ARGV.include?("--force-auto-update")
```

Comment:
This can be simplified to just:

```ruby
def force_auto_update?
  ARGV.include?("--force-auto-update")
end
```

Reply:
I actually did it this way on purpose so that it returns `nil` by default. More information in [this comment].

[this comment]: https://github.com/Homebrew/brew/pull/4064#discussion_r181530968

Reply:
Does my suggestion in https://github.com/Homebrew/brew/pull/4064#discussion_r181532198 help resolve this? Having a predicate method return `nil` is a little unintuitive, especially since `:force_auto_update` gets documented below as being a Boolean.

Reply:
Unfortunately not. How would you recommend documenting it as an optional boolean?

Reply:
Pushed a minor change that adds documentation to this function and changes the description of `:force_auto_update` to read as follows:
```ruby
# @option options [Boolean] :force_auto_update If present, whether to override the logic
#  that skips non-GitHub repositories during auto-updates.
```
Thoughts?

Reply:
That looks good, although you should turn `[Boolean]` into `[Boolean, nil]` so that YARD can mark it as nilable.

Reply:
Done.d